hooks/pre/30-upgrade.rb

```diff
@@ -102,6 +102,36 @@ def mark_qpid_cert_for_update
   end
 end
 
+# rubocop:disable MethodLength
+# rubocop:disable Style/MultipleComparison
+def upgrade_qpid_paths
+  qpid_dir = '/var/lib/qpidd'
+  qpid_data_dir = "#{qpid_dir}/.qpidd"
+  qpid_linearstore = "#{qpid_data_dir}/qls"
+  if File.exist?("#{qpid_linearstore}/dat2") && File.exist?("#{qpid_linearstore}/p001/efp/2048k/in_use")
+    logger.info 'Qpid directory upgrade is already complete, skipping'
+  else
+    # Make new directory structure for migration
+    Kafo::Helpers.execute("mkdir -p #{qpid_linearstore}/p001/efp/2048k/in_use")
+    Kafo::Helpers.execute("mkdir -p #{qpid_linearstore}/p001/efp/2048k/returned")
+    Kafo::Helpers.execute("mkdir -p #{qpid_linearstore}/jrnl2")
+    # Copy dat dir to new location dat2
+    Kafo::Helpers.execute("cp -r #{qpid_linearstore}/dat #{qpid_linearstore}/dat2")
+    # Move qpid jrnl files
+    Dir.foreach("#{qpid_linearstore}/jrnl") do |item|
+      next if item == '.' || item == '..'
+      Kafo::Helpers.execute("puts Moving #{item}")
+      Kafo::Helpers.execute("mv #{qpid_dir}/#{item} #{qpid_linearstore}/p001/efp/2048k/in_use/#{item}")
+      Kafo::Helpers.execute("ln -s #{qpid_linearstore}/p001/efp/2048k/in_use/#{item} #{qpid_linearstore}/jrnl2/#{item}")
```

Comment:
These two lines need to be performed in a loop over the #{qpid_linearstore}/jrnl/#{item} directory - for every file in this directory, move it to `in_use` dir, and create a symlink to it. So something like:

    Dir.foreach("#{qpid_linearstore}/jrnl/#{item}") do |jrnlfile|
        Kafo::Helpers.execute("mv #{qpid_linearstore}/jrnl/#{item}/#{jrnlfile} #{qpid_linearstore}/p001/efp/2048k/in_use/#{jrnlfile}")
        Kafo::Helpers.execute("ln -s #{qpid_linearstore}/p001/efp/2048k/in_use/#{jrnlfile} #{qpid_linearstore}/jrnl2/#{item}/#{jrnlfile}")
    end